plugin/plugin.go

```diff
@@ -0,0 +1,288 @@
+package plugin
+
+import (
+	"context"
+	"encoding/xml"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+)
+
+// Args represents the plugin's configurable arguments.
+type Args struct {
+	ReportFilenamePattern     string `envconfig:"PLUGIN_REPORT_FILENAME_PATTERN"`
+	FailedFails               int    `envconfig:"PLUGIN_FAILED_FAILS"`
+	FailedSkips               int    `envconfig:"PLUGIN_FAILED_SKIPS"`
+	FailureOnFailedTestConfig bool   `envconfig:"PLUGIN_FAILURE_ON_FAILED_TEST_CONFIG"`
+	UnstableFails             int    `envconfig:"PLUGIN_UNSTABLE_FAILS"`
+	UnstableSkips             int    `envconfig:"PLUGIN_UNSTABLE_SKIPS"`
```

Comment:
let's remove inputs related to unstable as Harness doesn't support the pipeline status as unstable